feat(flutter-searchbox): add docs ref for state-provider widget (#302)
appbaseio/Docs: content/docs/reactivesearch/flutter-searchbox/stateprovider.md

---
title: 'StateProvider'
meta_title: 'API Reference for StateProvider'

> meta_description: 'StateProvider allows you to access the current state of your flutter app.'

Reviewer: ```suggestion
meta_description: 'StateProvider allows you to access the current state of flutter-searchbox widgets.'
```

keywords:
    - flutter-searchbox
    - api-reference
    - state-provider
    - elasticsearch
    - search-ui
sidebar: 'docs'
nestedSidebar: 'flutter-searchbox'
---

# How does it work?

[StateProvider](https://pub.dev/documentation/flutter_searchbox/latest/flutter_searchbox/StateProvider-class.html) allows you to access the current state of your app widgets along with the search results. For instance, you can use this component to create results/no results or query/no query pages.

Example Use(s):
 - perform side-effects based on the results states of various widgets.
 - render custom UI based on the current state of app.

## Usage

### Basic Usage

```dart
StateProvider(
  ...

>   onChange: (prevState, nextState) {

Reviewer: ```suggestion
  onChange: (nextState, prevState) {
```

        // do something here
    },
  ...
)
```

### Usage with All Props

```dart
StateProvider(
  subscribeTo: {
   'result-component': [KeysToSubscribe.Results, KeysToSubscribe.From]
  },       

>   onChange: (prevState, nextState) {

Reviewer: ```suggestion
  onChange: (nextState, prevState) {
```

    // ensure all routes have access to the store.
    return SearchBaseProvider(
      // Pass the searchbase instance to the SearchBaseProvider. Any ancestor `SearchWidgetConnector`
      // widgets will find and use this value as the `SearchController`.
      searchbase: searchbaseInstance,
      child: MaterialApp(
        title: "SearchBox Demo",
        theme: ThemeData(
          primarySwatch: Colors.blue,
          visualDensity: VisualDensity.adaptivePlatformDensity,
        ),
        home: HomePage(),
      ),
    );
  }
}

class HomePage extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'SearchBox Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: Scaffold(
          bottomNavigationBar: Padding(
            padding: EdgeInsets.all(20.0),
            child: StateProvider(
              subscribeTo: {
                "result-widget": [KeysToSubscribe.Results , KeysToSubscribe.Value, KeysToSubscribe.From],                
              },

>               onChange: (prevState, nextState) {

Reviewer: ```suggestion
              onChange: (nextState, prevState) {
```
